### ISSUE TITLE
Server crash on EntityMagicProjectile.writeSpawnData()

### ISSUE THREAD

**Issue by LemADEC** [Electroblob77/Wizardry#71]
Minecraft version: 1.12.2
Wizardry version: 4.1.2
Environment: Server

Issue details:
Server crashed

Other mods involved:
n/a

Link to crash report (if applicable):
```
[19:39:18] [Server thread/ERROR] [minecraft/MinecraftServer]: Encountered an unexpected exception
io.netty.handler.codec.EncoderException: java.lang.NullPointerException
>       at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:106) ~[minecraft_server.1.12.2.jar:?]
>       at io.netty.handler.codec.MessageToMessageCodec.write(MessageToMessageCodec.java:116) ~[minecraft_server.1.12.2.jar:?]
>       at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:738) ~[minecraft_server.1.12.2.jar:?]
>       at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:730) ~[minecraft_server.1.12.2.jar:?]
>       at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:816) ~[minecraft_server.1.12.2.jar:?]
>       at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:723) ~[minecraft_server.1.12.2.jar:?]
>       at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:704) ~[minecraft_server.1.12.2.jar:?]
>       at io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1017) ~[minecraft_server.1.12.2.jar:?]
>       at io.netty.channel.AbstractChannel.write(AbstractChannel.java:286) ~[minecraft_server.1.12.2.jar:?]
>       at io.netty.channel.embedded.EmbeddedChannel.writeOutbound(EmbeddedChannel.java:341) ~[minecraft_server.1.12.2.jar:?]
>       at net.minecraftforge.fml.common.network.FMLEmbeddedChannel.generatePacketFrom(FMLEmbeddedChannel.java:72) ~[FMLEmbeddedChannel.class:?]
>       at net.minecraftforge.fml.common.network.internal.FMLNetworkHandler.getEntitySpawningPacket(FMLNetworkHandler.java:134) ~[FMLNetworkHandler.class:?]
>       at net.minecraft.entity.EntityTrackerEntry.createSpawnPacket(EntityTrackerEntry.java:501) ~[os.class:?]
>       at net.minecraft.entity.EntityTrackerEntry.updatePlayerEntity(EntityTrackerEntry.java:380) ~[os.class:?]
>       at net.minecraft.entity.EntityTrackerEntry.updatePlayerEntities(EntityTrackerEntry.java:490) ~[os.class:?]
>       at net.minecraft.entity.EntityTrackerEntry.updatePlayerList(EntityTrackerEntry.java:143) ~[os.class:?]
>       at net.minecraft.entity.EntityTracker.tick(EntityTracker.java:292) ~[ol.class:?]
>       at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:779) ~[MinecraftServer.class:?]
>       at net.minecraft.server.dedicated.DedicatedServer.updateTimeLightAndEntities(DedicatedServer.java:397) ~[nz.class:?]
>       at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:668) ~[MinecraftServer.class:?]
>       at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
>       at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]
Caused by: java.lang.NullPointerException
>       at electroblob.wizardry.entity.projectile.EntityMagicProjectile.writeSpawnData(EntityMagicProjectile.java:111) ~[EntityMagicProjectile.class:?]
>       at electroblob.wizardry.entity.projectile.EntityBomb.writeSpawnData(EntityBomb.java:39) ~[EntityBomb.class:?]
>       at net.minecraftforge.fml.common.network.internal.FMLMessage$EntitySpawnMessage.toBytes(FMLMessage.java:213) ~[FMLMessage$EntitySpawnMessage.class:?]
>       at net.minecraftforge.fml.common.network.internal.FMLRuntimeCodec.encodeInto(FMLRuntimeCodec.java:42) ~[FMLRuntimeCodec.class:?]
>       at net.minecraftforge.fml.common.network.internal.FMLRuntimeCodec.encodeInto(FMLRuntimeCodec.java:32) ~[FMLRuntimeCodec.class:?]
>       at net.minecraftforge.fml.common.network.FMLIndexedMessageToMessageCodec.encode(FMLIndexedMessageToMessageCodec.java:81) ~[FMLIndexedMessageToMessageCodec.class:?]
>       at io.netty.handler.codec.MessageToMessageCodec$1.encode(MessageToMessageCodec.java:67) ~[minecraft_server.1.12.2.jar:?]
>       at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:88) ~[minecraft_server.1.12.2.jar:?]
>       ... 21 more
```

Nothing in the log in the 3 previous seconds, so I'm not sure what triggered it.

**Comment by Grekkor** [comment]
I have the same issue too. Server regularly crashes due to this.
Here my full Server crashreport:
https://pastebin.com/z1MLycKh

**Comment by LemADEC** [comment]
@Crusher565 your crash is about `Caused by: java.lang.IllegalArgumentException: Can not set boolean field net.minecraft.entity.EntityLivingBase.field_70703_bu to java.lang.Float` which seems unrelated to the original one. Check other issues or make a new ones for yours.

**Comment by Electroblob77** [comment]
@LemADEC Looks like a projectile has lost its caster, probably because they logged out. I thought I'd caught all of these but apparently not, I'll look into it. Thanks for reporting.

@Crusher565 I've done a bit of digging and as I suspected, wizardry's involvement in that crash report is a complete coincidence (as you said yourself on cyberware's issue tracker, it happens regardless of whether wizardry is installed or not). This *was* a bug in Cyberware, which has apparently been fixed in the latest version, 0.2.10.15, so I suggest you update to that. Link to Cyberware issue, for reference: https://github.com/An-Sar/Cyberware/issues/65
